spec/services/replication/shared_examples_provider.rb

```diff
@@ -44,7 +44,8 @@
     describe '::Aws::S3::Object#upload_file' do
       subject(:s3_object) { bucket.object("test_key_#{test_key_id}") }
 
-      let(:test_key_id) { ENV.fetch('CIRCLE_SHA1', '000')[0..6] }
```

Comment:
the key was previously based on the sha of the commit being tested, making the key the same between CI runs on the same commit.  there is still one occasionally flappy test, so then when that one succeeds on retry, this one is sure to fail, unless something is done to cause a commit hash change.